### PR TITLE
Updating the noodlepad_micro keymaps with custom keycodes

### PR DESCRIPTION
**Pull request by jessel92**
## Description
I added some custom keycodes to the noodlepad_micro keymaps. This also required me to modify the config,h file with the features implemented in the keycodes. 

Forgive me if there is json version of some of the config.h options I enabled, I tried some and they didn't work, and I'm not really sure where to find if there is. Let me know! 

Also, I'm sure the keycodes code is a little messy and inefficient, but they work great hahaha

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by jessel92** [comment]
Any movement on this?

File: keyboards/themadnoodle/noodlepad_micro/config.h
Comment (drashna) [inline keyboards/themadnoodle/noodlepad_micro/config.h:10]: These should be moved to info.json.  `qmk info -f json` can help with this.

File: keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c
Comment (drashna) [inline keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c:13]: `QK_KB_0` should be used at the kb level, not the keymap level. 
```suggestion
    L_IND = SAFE_RANGE, // Toggle the Layer Indicators Modes
    L_CYC               // Cycle through the layers
```

File: keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c
Comment (drashna) [inline keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c:1]: default and via keymaps should be simple and pristine.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/themadnoodle/noodlepad_micro/config.h
Comment (jessel92) [reply keyboards/themadnoodle/noodlepad_micro/config.h:10]: I had tried to add these to the JSON but they didn't work that's why they are in the config. So wasn't sure if they were implemented into to the JSON yet. 
If they are. Can you show me how to add them? 
All the ways I tried failed. 

If you don't mind be asking what does the `qmk info -f json` command do? I tried to use both `qmk info -f json -kb themadnoodle/noodlepad_micro`  and `qmk info -f json -kb themadnoodle/noodlepad_micro -km via` but nothing changed.

File: keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c
Comment (jessel92) [reply keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c:13]: Also, this was something that took a bunch of work to figure out with the help of the QMK Discord. This was the only way VIA would recognize the custom keycodes. 

It was a while ago, but I believe I tried to put it at the KB level, but it wasn't recognized by VIA either. Well, it would get recognized, but it wouldn't show up as the custom code. It would show up as another name, and i couldn't map to a switch. 

To help a noob out, should I remove everything below '/*Encoder Mapping*/' section and put it in a noodlepad_mico.c file along with the 'enum custom_keycodes' at the top?

If you can help me out, that'd be amazing because I'm really at a total loss.

**Comment by jessel92** [comment]
@drashna 
I'm also now getting THIS error that i was not getting before when i try to compile my keymap

```
/RP2040/ws2812_vendor.c:4:
drivers/ws2812.h:60:30: error: 'RGBLIGHT_LED_COUNT' undeclared here (not in a function); did you mean 'RGBLIGHT_LAYER_BLINK'?
   60 | #    define WS2812_LED_COUNT RGBLIGHT_LED_COUNT
      |                              ^~~~~~~~~~~~~~~~~~
platforms/chibios/drivers/vendor/RP/RP2040/ws2812_vendor.c:138:46: note: in expansion of macro 'WS2812_LED_COUNT'
  138 | static uint32_t                WS2812_BUFFER[WS2812_LED_COUNT];
      |                                              ^~~~~~~~~~~~~~~~
platforms/chibios/drivers/vendor/RP/RP2040/ws2812_vendor.c:138:32: error: 'WS2812_BUFFER' defined but not used [-Werror=unused-variable]
  138 | static uint32_t                WS2812_BUFFER[WS2812_LED_COUNT];
      |                                ^~~~~~~~~~~~~
cc1.exe: all warnings being treated as errors
 [ERRORS]
 |
 |
 |
make: *** [builddefs/common_rules.mk:373: .build/obj_themadnoodle_noodlepad_micro_via/ws2812_vendor.o] Error 1
```

File: keyboards/themadnoodle/noodlepad_micro/config.h
Comment (jessel92) [reply keyboards/themadnoodle/noodlepad_micro/config.h:10]: This what i get when i try to add those to the JSON

```
☒ Not including data from file: keyboards/themadnoodle/noodlepad_micro/info.json
☒       rgblight: Additional properties are not allowed ('layer_blink', 'layers_override_rgb', 'max_layers' were unexpected)
☒ 'matrix_size'
Traceback (most recent call last):
  File "C:/QMK_MSYS/mingw64/lib/python3.10/site-packages/milc/milc.py", line 528, in __call__
    return self.__call__()
  File "C:/QMK_MSYS/mingw64/lib/python3.10/site-packages/milc/milc.py", line 533, in __call__
    return self._subcommand(self)
  File "C:/madnoodlefirmware/qmk_firmware/lib/python/qmk/cli/generate/rules_mk.py", line 65, in generate_rules_mk
    kb_info_json = dotty(info_json(cli.args.keyboard))
  File "C:/madnoodlefirmware/qmk_firmware/lib/python/qmk/info.py", line 221, in info_json
    info_data = _extract_led_config(info_data, str(keyboard))
  File "C:/madnoodlefirmware/qmk_firmware/lib/python/qmk/info.py", line 717, in _extract_led_config
    cols = info_data['matrix_size']['cols']
KeyError: 'matrix_size'
```

File: keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c
Comment (drashna) [reply keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c:13]: default keymap shouldn't have via enabled.

Comment (jessel92) [reply keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c]: @drashna AH I see! So you're saying just for the default keymap.

File: keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c
Comment (jessel92) [reply keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c:1]: @drashna, things should be good to go, as far as I can tell.

**Comment by jessel92** [comment]
@drashna Please let me know if we can get this merged as soon as possible. Because I have a PR with VIA that predates these custom codes, and they kindly agreed to wait for these changes to be merged before proceeding with their PR.

**Review by drashna — APPROVED** [review]

**Comment by jessel92** [comment]
@drashna, I see everything was approved, but the PR was never merged. Is there any way we can get this merged?

**Comment by waffle87** [comment]
At least two review approvals are needed before merge.

**Review by fauxpark — COMMENTED** [review]
Keyboard updates need to go through develop.

File: keyboards/themadnoodle/noodlepad_micro/config.h
Comment (waffle87) [inline keyboards/themadnoodle/noodlepad_micro/config.h:10]: These should be migrated to `info.json`:
```json
"rgblight": {
    "layers": {
        "enabled": true,
        "blink": true,
        "max": 8,
        "override_rgb": false
    }
}
```
```suggestion
```

File: keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c
Comment (waffle87) [inline keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c:7]:
```suggestion
    L_IND = QK_USER, // Toggle the Layer Indicators Modes
```

File: keyboards/themadnoodle/noodlepad_micro/keymaps/via/keymap.c
Comment (waffle87) [inline keyboards/themadnoodle/noodlepad_micro/keymaps/via/keymap.c:11]:
```suggestion
    L_IND = QK_USER,
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by waffle87** [comment]
Also, see merge conflicts to be resolved.

**Comment by jessel92** [comment]
> At least two review approvals are needed before merge.

@waffle87  Ahh, I see ok, thanks for the info.
@fauxpark This is a keymap update, isn't it? So shouldn't it be able to be merged straight into master??

**Comment by fauxpark** [comment]
There are changes to keyboard-level files.

**Comment by jessel92** [comment]
> There are changes to keyboard-level files.

@fauxpark So, does that mean I need to wait until May for it to be merged into the master branch from develop, just because I added a few config changes?

Also, what's this branch conflict about? It wasn't there a few minutes ago, and how do I resolve it?

**Comment by waffle87** [comment]
> does that mean I need to wait until May for it to be merged into the master branch from develop
Yes. The merge conflicts are product of switching the target branch. See [this doc](https://docs.qmk.fm/#/newbs_git_resolving_merge_conflicts).

**Comment by jessel92** [comment]
@waffle87 ok, I think I was able to resolve it all. Correct? I rebased to develop, then recommitted the changes made in the master branch.

File: keyboards/themadnoodle/noodlepad_micro/config.h
Comment (fauxpark) [inline keyboards/themadnoodle/noodlepad_micro/config.h:6]: Keep the newline at EOF here please.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/themadnoodle/noodlepad_micro/config.h
Comment (jessel92) [reply keyboards/themadnoodle/noodlepad_micro/config.h]: I added a newline to the end of the file is that what you mean?

Also, can we get this resolved and merged as soon as we can this has been marked as stale and will be closed even though I've been making the changes as soon as they are requested of me.

**Comment by jessel92** [comment]
@fauxpark @waffle87 @drashna

This PR has been open since February 2, meaning it has now been pending for six months. I have promptly addressed every requested change within hours of the request. However, GitHub has attempted to mark this as stale twice, even though I am merely waiting for the next steps from everyone involved. Considering that this now ALSO needs to go through the develop branch before hitting the master, I would really appreciate it if we could resolve this as soon as possible (unless there is a way for us to get this merged directly to the master branch).

I am uncertain what else I need to do.

Additionally, due to the extremely long wait time, this PR has become somewhat outdated. I have new changes that need to be added once this is merged, along with a new keypad PR that I can't open until this is incorporated.

I am trying to remain patient, understanding that you all have a lot on your plate and that open-source work can be challenging. However, this prolonged process has significantly thrown a wrench in my business at this point.

File: keyboards/themadnoodle/noodlepad_micro/rules.mk
Comment (fauxpark) [inline keyboards/themadnoodle/noodlepad_micro/rules.mk:1]: This file should not be re-added. Please delete it.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/themadnoodle/noodlepad_micro/rules.mk
Comment (jessel92) [reply keyboards/themadnoodle/noodlepad_micro/rules.mk:1]: Removed

**Comment by tzarc** [comment]
As of August 26, 2024, `qmk/qmk_firmware` is no longer accepting VIA-enabled keymaps as these have now transitioned to a repository under the VIA team's control.

As you've submitted a PR containing `via` or VIA-enabled keymap(s), this is your notice that they should be removed from this PR. You should now submit a secondary PR to the [VIA QMK Userspace](https://github.com/the-via/qmk_userspace_via) repository with your associated `via` or VIA-enabled keymaps instead.

**Pull request closed without merging**
